### PR TITLE
Generate the manifest secret with YAML

### PR DESCRIPTION
**Pull request by pfmackin**
# Description

1. Generate the manifest secret that has the YAML to create the registration secret
2. Rename sync files to be consistent
3. Fix unit test and add to integ test

Fixes VZ-2124

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/clusters/sync_manifest.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+// Create or update a secret that contains Kubernetes resource YAML which will be applied
+// on the managed cluster to create resources needed there. The YAML has multiple Kubernetes
+// resources separated by 3 hyphens ( --- ), so that applying the YAML will create/update multiple
+// resources at once.  This YAML is stored in the Verrazzano manifest secret.
+func (r *VerrazzanoManagedClusterReconciler) syncManifestSecret(vmc *clusterapi.VerrazzanoManagedCluster) error {
+	const (
+		yamlSep                           = "---\n"
+		targetRegistrationSecretName      = "verrazzano-cluster"
+		targetRegistrationSecretNamespace = "verrazzano-system"
```
Comment (desagar) [inline platform-operator/controllers/clusters/sync_manifest.go:29]: there's a constant for these on the application-operator side - I'd suggest adding one on platform operator side if there isn't one. I know Mike has been insisting we don't strew these assumptions all over the place

File: platform-operator/test/integ/integ_test.go [hunk 2]
```diff
@@ -156,3 +166,17 @@ func verifyClusterSecret() {
...
+
+// Verify the manifest secrets
+func verifyManifestSecret() {
+	secret, err := K8sClient.GetSecret(vzclusters.GetManifestSecretName(managedClusterName), vzMcNamespace)
+	if err != nil {
+		ginkgo.Fail(fmt.Sprintf("Unable to get cluster secret %s that contains kubeconfig: %v", vzclusters.GetManifestSecretName(managedClusterName), err))
+	}
+
+	// Get the yaml from the secret
+	kubconfigBytes := secret.Data["yaml"]
```
Comment (desagar) [inline platform-operator/test/integ/integ_test.go:178]: minor point but this is not just kubeconfigBytes anymore now

Comment (desagar) [reply platform-operator/test/integ/integ_test.go:178]: Also I'm thinking maybe we should try to unmarshall this YAML to make sure it's valid, at a minimum

File: platform-operator/controllers/clusters/sync_manifest.go (hunk 1 above)
Comment (pfmackin) [reply platform-operator/controllers/clusters/sync_manifest.go:29]: i think the scope of a constant you be a limited as possible (like variables)  Only public package level constants should go in a constants file

File: platform-operator/test/integ/integ_test.go (hunk 2 above)
Comment (pfmackin) [reply platform-operator/test/integ/integ_test.go:178]: right i will fix post merge

File: platform-operator/controllers/clusters/sync_manifest.go (hunk 1 above)
Comment (desagar) [reply platform-operator/controllers/clusters/sync_manifest.go:29]: verrazzano-system is one such, isn't it?

Comment (desagar) [reply platform-operator/controllers/clusters/sync_manifest.go:29]: and verrazzano-cluster should probably go in at least a localized constant

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**